test/jwt.test.js

```diff
@@ -0,0 +1,341 @@
+};
+
+function generateJWT(bodyOverrides, alg){
+  var body = Object.assign({}, payload, bodyOverrides || {});
+  alg = alg || 'RS256';
+  var options = {
+    header: { alg: alg },
+    payload: body
+  };
+  if (alg === 'RS256') {
+    options.privateKey = privateKeyRSA;
+  } else if (alg === 'HS256') {
+    options.secret = secretHMAC;
+  }
+  return jws.sign(options);
+}
+
+describe('jwt.decode', function() {
+  it('should decode a valid token', function() {
+    var alg = 'RS256';
+    var token = generateJWT({}, alg);
+    var decoded = jwt.decode(token);
+
+    assert.equal(decoded._raw, token);
+    assert.equal(decoded.header.alg, alg);
+
+    Object.keys(payload).forEach(function(key) {
+      assert.deepEqual(payload[key], decoded.payload[key])
+    })
+  });
+
+  it('throws errors on invalid tokens', function() {
+    var IDTOKEN_ERROR_MESSAGE = 'ID token could not be decoded';
+
+    it('throws when there is less than 3 parts', function() {
+      assert.throws(jwt.decode('test'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(jwt.decode('test.'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(jwt.decode('test.test'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(jwt.decode('test.test.test.test'), 
+        IDTOKEN_ERROR_MESSAGE
+      );
+    });
+    it('throws when there is no header', function() {
+      assert.throws(jwt.decode('.test.test'), IDTOKEN_ERROR_MESSAGE);
+
+    });
+    it('throws when there is no payload', function() {
+      assert.throws(jwt.decode('test..test'), IDTOKEN_ERROR_MESSAGE);
+
+    });
+    it('throws when there is no signature', function() {
+      assert.throws(jwt.decode('test.test.'), IDTOKEN_ERROR_MESSAGE);
+
+    });
+  })
+});
+
+describe('jwt.verify', function() {
+  var expectedOptions;
+
+  beforeEach(function() {
+    expectedOptions = Object.assign({}, payload);
+    expectedOptions.aud = expectations.clientId;
+    expectedOptions.max_age = tomorrow();
+  });
+
+  it('should throw when no id token is present', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'ID token is required but missing';
+    try {
+      jwt.verify()
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when no Issuer is present in the payload', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'Issuer (iss) payload must be a string present in the ID token';
+    try {
+      jwt.verify(generateJWT({ iss: undefined }))
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when the expected issuer is not in the payload', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'Issuer (iss) payload mismatch in the ID token; expected "DifferentIssuer", found "https://tokens-test.auth0.com/"';
```

Comment:
```suggestion
    var EXPECTED_ERROR_MESSAGE = 'Issuer (iss) payload mismatch in the ID token; expected "ExpectedIssuer", found "https://tokens-test.auth0.com/"';
```

Makes this a little more clear ... took me a bit to puzzle through the test here